### ISSUE TITLE
Improve TableSymbol/datashape documentation

### ISSUE THREAD

**Issue by wrobstory** [ContinuumIO/blaze#1006]
I'm working through the docs/tutorial, and am finding the semantics of what I can use in a TableSymbol schema frustrating. 

Things that seem strange/aren't obvious: 
- `string` works, but `str` doesn't
- `float32` works, but `float` doesn't
- `int` works, but `integer` doesn't
- `bool` works, but `boolean` doesn't

IMO, all of the base Python types (`float`, `str`, `int`) should work out of the box.

**Comment by mrocklin** [comment]
Indeed, the fact that you've actually run into `TableSymbol` in the wild is somewhat discouraging, that interface is fairly old.  Can you point me to the docs/tutorial that you're using?

What are you doing that you have to specify a datashape?

**Comment by wrobstory** [comment]
hehehehe- I was using your blog post here: http://matthewrocklin.com/blog/work/2014/09/01/Blaze-Expressions/

Based on your blog posts and the docs, this is what I put together: http://nbviewer.ipython.org/github/wrobstory/pdxpython2015/blob/master/PDXPython2015.ipynb#Blaze

I'd be really interested to get feedback on what I should be doing there instead of using TableSymbol (or if there are any cool features I should talk about). I still have time to change it before I talk to a group of devs about Blaze :grin:

**Comment by cpcloud** [comment]
There are a few things to discuss around the mapping of python types to datashape types. Datashape types generally follow C conventions (though there may be some inconsistencies there), whereas Python follows its own conventions. `float` is typically 32 bits in C and in Python it's backed by C's `double` which is typically 64 bits. `int` in datashape maps to `int32` according to the typical precision of C `int`.

I'd be +1 on the following:

* `str` -> `string`
* `float` -> `float32`

Inside of the Python implementation of datashape are generic type classes `integral` and `boolean` both of which are not designed to be used directly (though a case *could* be made for `boolean`, but personally i don't see the benefit of that).

```
In [1]: import datashape

In [2]: datashape.boolean
Out[2]: {boolean}

In [3]: type(datashape.boo
datashape.bool_    datashape.boolean

In [3]: type(datashape.boolean)
Out[3]: datashape.typesets.TypeSet

In [5]: datashape.integral
Out[5]: {integral}

In [6]: datashape.integral.types
Out[6]:
(ctype("int8"),
 ctype("uint8"),
 ctype("int16"),
 ctype("uint16"),
 ctype("int32"),
 ctype("uint32"),
 ctype("int64"),
 ctype("uint64"))
```

If none of my above suggestions are controversial, then I'm happy to implement them in datashape.

**Comment by cpcloud** [comment]
@wrobstory instead of `TableSymbol` the preferred way to create *any* kind of `Symbol` is through the `symbol` function. `TableSymbol` simply adds a `var` dimension to a passed in `Record` datashape. Here's how you'd do it with your `diamonds` schema:

```python
In [8]: bz_cols = ('var * {carat: float64, cut: string, color: string, clarity: string, '
           'depth: float64, table: float64, price: int64, x: float64, y: float64, z: float64}')

In [9]: from blaze import symbol

In [10]: diamonds = symbol('diamonds', bz_cols)

In [11]: diamonds.dshape
Out[11]:
dshape("""var * {
  carat: float64,
  cut: string,
  color: string,
  clarity: string,
  depth: float64,
  table: float64,
  price: int64,
  x: float64,
  y: float64,
  z: float64
  }""")
```

**Comment by cpcloud** [comment]
Also, for interactive usage `Data` is a lot easier to use. Docs [here](http://blaze.pydata.org/en/latest/quickstart.html#blaze-interactive-data). `symbol`s are nice though when trying to explain how blaze separates expressions from data

**Comment by cpcloud** [comment]
@wrobstory one last comment :)

Regarding these lines:

```python
pg_datasource = bz.odo(df_diamonds, "postgresql://postgres:postgres@localhost/pydata::diamonds", 
                       dshape=diamonds.dshape)
[x for x in bz.compute(carat_count, pg_datasource).execute()]
```

You don't have to explicitly call `execute` on the result of `compute`, `odo` can do this for you:

```python
df = odo(compute(carat_count, pg_datasource), pd.DataFrame)
```

**Comment by cpcloud** [comment]
Alright I'm just going to keep commenting :)

When creating the `diamonds` symbol you don't have to specify a `dshape`
```
```

**Comment by mrocklin** [comment]
Given the example I would probably stress the following workflow

```python
>>> diamonds = bz.Data("postgresql://postgres:postgres@localhost/pydata::diamonds")
>>> carat_count = bz.by(diamonds[diamonds.price > 1000].clarity,
...                     carat=diamonds.carat.count())
>>> list(carat_count)  
...

or just 

>>> carat_count
...
```

You've been playing with the internal, explicit API which, while interesting to those present, is usually a bit mind-bending to your usual meetup audience.  There was a decent amount of work to make sure that the interface around dealing with a database felt a lot like dealing with a dataframe or numpy array.  Explicit calls to compute and dshape kwargs detract from this.

**Comment by wrobstory** [comment]
Thanks to you both- I definitely want to stress how powerful the idea of separating expressions from data sources is. I think it's probably the most interesting/important bit of working with Blaze. 

Again, thank you!
